cadc-storage-permissions/src/main/java/org/opencadc/inventory/permissions/ReadGrant.java

```diff
@@ -0,0 +1,107 @@
+ *  written permission.                  préalable et particulière
+ *                                       par écrit.
+ *
+ *  This file is part of the             Ce fichier fait partie du projet
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  $Revision: 4 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.inventory.permissions;
+
+import java.net.URI;
+import java.util.Date;
+import org.opencadc.inventory.InventoryUtil;
+
+/**
+ * Holds read grant information about an artifact.
+ * 
+ * @author majorb
+ *
+ */
+public class ReadGrant extends Grant {
+
+    // Is the artifact available to anonymous (all) users.
+    private boolean anonymousAccess;
+
+    /**
+     * Construct a read grant for the given artifactURI and expiry date.
+     * @param artifactURI The applicable targetURI.
+     * @param expiryDate The expiry date of the grant.
+     * @param anonymousAccess true is the artifact has anonymous access, false otherwise.
+     */
+    public ReadGrant(URI artifactURI, Date expiryDate, boolean anonymousAccess) {
+        super(artifactURI, expiryDate);
+        InventoryUtil.assertNotNull(ReadGrant.class, "anonymousAccess", anonymousAccess);
```

Comment:
You can remove this check because primitives can't be null.

Reply:
Just the one change above... otherwise looks good.

Reply:
Done